core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -200,13 +202,14 @@ private CloseableIterable<? extends ContentFile<?>> files(Schema fileProjection)
       }
     }
 
-    private StructLike withReadableMetrics(ContentFile<?> file) {
-      int expectedSize = projection.columns().size();
+    private StructLike withReadableMetrics(ContentFile<?> file, int metricsPosition) {
+      int columnSize = projection.columns().size();
```

Comment:
nit: `fieldCount` or `columnCount` is probably more clear. `columnSize` can be read as the size of one particular column.